www/js/coolhole-callbacks.js

```diff
@@ -31,7 +31,11 @@ const CoolholeCallbacks = {
     CHANNEL.usersCoolPoints = response.data;
 
     // Update the UI
+    // Check the users rank and fade in the counter/button
     initPointsForSelf(CLIENT.coolpoints);
+    if (CLIENT.rank >= 1) {
+      initCounterForSelf();
+    }
```

Comment:
nit: Do we need these to be separated? Also I plan on trying to tackle (more broadly) the issue of non-logged in or ineligible users from getting the `init` callback hit:
https://github.com/streeeggs/coolhole/pull/22/files#diff-9b2696fd14264957ce69042fbba8e415728fb13046d05b9c2d213165526b99d4R271-R272

That being said if we needed these separated and were concerned with a client-side check , I would think it would be a concern both `initPointsForSelf` and `initCounterForSelf`

Reply:
I think that was the main reason I made them separate in the first place. If the user is logged in you get the points counter and button, logged out it is just the button. That way anons and logged out users can still see the modal.
I thought about a button for non logged in users, but not having the counter start seemed a simpler solution.
I can try out some ideas on your branch to see what can be done with the check in place thought.

Reply:
Yea I see. I would say you should, instead, decouple the button rendering from this callback function. I would suggest adding the "counter" if the user is valid but always try to render the button outside of this callback. Maybe at the top of `coolpoints-utils.js`